src/NLog/Targets/DatabaseParameterInfo.cs

```diff
@@ -98,6 +98,20 @@ public DatabaseParameterInfo(string parameterName, Layout parameterLayout)
         /// <docgen category='Parameter Options' order='10' />
         [DefaultValue(0)]
         public byte Scale { get; set; }
+
+        /// <summary>
+        /// Gets or sets the column type of the database.
+        /// </summary>
+        /// <docgen category='Parameter Options' order='10' />
+        [DefaultValue(null)]
+        public string DbType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the database column type property.
+        /// </summary>
+        /// <docgen category='Parameter Options' order='10' />
+        [DefaultValue(null)]
+        public string DbTypePropertyName { get; set; }
```

Comment:
Oracle uses a different property on its object for its custom database type then the property implemented by IDbDataParameter. Perhaps there is a better way to figure out the property we need to set. One thought I have is we could use reflection to find a property that matches what ever DbType is set to then set the value to that property. Do you have any suggestions?